interpreter/beam/beam.go

```diff
@@ -231,11 +354,191 @@ func (i *beamInstance) Symbolize(frame *host.Frame, frames *libpf.Frames) error
 	if !frame.Type.IsInterpType(libpf.BEAM) {
 		return interpreter.ErrMismatchInterpreterType
 	}
+	codeHeader := libpf.Address(frame.File)
+	pc := libpf.Address(frame.Lineno)
 
+	log.Debugf("BEAM symbolizing pc: 0x%x", pc)
+
+	functionIndex, moduleID, functionID, arity, err := i.findMFA(pc, codeHeader)
+	if err != nil {
+		return err
+	}
+
+	moduleName, err := i.lookupAtom(moduleID)
+	if err != nil {
+		return err
+	}
+	functionName, err := i.lookupAtom(functionID)
+	if err != nil {
+		return err
+	}
+
+	mfaName := ""
+	if strings.HasPrefix(moduleName, "Elixir.") {
+		// This is an Elixir module, so format the function using Elixir syntax (without the "Elixir." prefix)
+		mfaName = fmt.Sprintf("%s.%s/%d", moduleName[7:], functionName, arity)
+	} else {
+		// Assume it's Erlang and format it using Erlang syntax
+		mfaName = fmt.Sprintf("%s:%s/%d", moduleName, functionName, arity)
+	}
```

Comment:
How often the `mfaName` here gets constructed? The top level will cache the frame, but this can get called for difference PC offsets a lot I suppose? Do you think it would make sense to add LRU for `findMFA` and also cache the `mfaName` there?